privacyidea/lib/eventhandler/tokenhandler.py

```diff
@@ -312,7 +319,66 @@ def actions(cls):
                                 "description": _("Remove a tokengroup from the token."),
                                 "value": [tg.name for tg in get_tokengroups()]
                             }
-                        }
+                        },
+                   ACTION_TYPE.ATTACH_APPLICATION:
+                       {"machine ID":
+                            {
+                                "type": "string",
+                                "required": True,
+                                "description": _("The ID of the machine you want to attach the token to")
```

Comment:
offline does not attach to a specific token anymore.
With offline, this can be left empty and it then will be connected to "any machine".

So it should not be required=True.

Reply:
@jona-samuel please fill in the answers in regards to my comments.